plasmapy/dispersion/numerical/hollweg_.py

```diff
@@ -0,0 +1,339 @@
+
+    ValueError
+        If ``k`` is negative or zero.
+
+    ValueError
+        If ``ion`` is not of category ion or element.
+
+    ValueError
+        If ``B``, ``n_i``, ``T_e``, or ``T_I`` are not single valued
+        `astropy.units.Quantity` (i.e. an array).
+
+    ValueError
+        If ``k`` or ``theta`` are not single valued or a 1-D array.
+
+    Warns
+    -----
+    : `~plasmapy.utils.exceptions.PhysicsWarning`
+        When the computed wave frequencies violate the
+        :math:`\omega/\omega_{\rm ci} \ll 1` assumption of the
+        dispersion relation.
+
+    Notes
+    -----
+
+    The equation presented in Hollweg 1999 [2]_ (equation 3 in Bellan
+    2012 [1]_) is:
+
+    .. math::
+        \left( \frac{\omega^2}{k_{\rm z}^2 v_{\rm A}^2} - 1 \right) &
+        \left[
+            \omega^2 \left( \omega^2 - k^2 v_{\rm A}^2 \right)
+            - \beta k^2 v_{\rm A}^2 \left(
+                \omega^2 - k_{\rm z}^2 v_{\rm A}^2
+            \right)
+        \right] \\
+        &= \omega^2 \left(\omega^2 - k^2 v_{\rm A}^2 \right) k_{\rm x}^2
+        \left(
+            \frac{c_{\rm s}^2}{\omega_{\rm ci}^2}
+            - \frac{c^2}{\omega_{\rm pe}^2} \frac{\omega^2}{k_{\rm z}^2v_{\rm A}^2}
+        \right)
+
+    where
+
+    .. math::
+        k_{\rm x} = \mathbf{k} \cdot \hat{x}
+
+    References
+    ----------
+    .. [1] PM Bellan, Improved basis set for low frequency plasma waves,
+       2012, JGR, 117, A12219, doi: `10.1029/2012JA017856
+       <https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2012JA017856>`_.
+
+    .. [2] JV Hollweg, Kinetic Alfven wave revisited, 1999, JGR, 104(A7),
+       14811–14819, doi: `10.1029/1998JA900132
+       <https://agupubs.onlinelibrary.wiley.com/doi/abs/10.1029/1998JA900132>`_
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> from plasmapy.dispersion.numerical import hollweg_
+    >>> inputs = {
+    ...    "k": np.logspace(-7, -2, 2) * u.rad / u.m,
+    ...    "theta": 88 * u.deg,
+    ...    "n_i": 5 * u.cm ** -3,
+    ...    "B": 2.2e-8 * u.T,
+    ...    "T_e": 1.6e6 * u.K,
+    ...    "T_i": 4.0e5 * u.K,
+    ...    "ion": Particle("p+"),
+    ... }
+    >>> omegas = hollweg(**inputs)
+    >>> omegas
+    {'fast_mode': <Quantity [2.62911663e-02, 2.27876968e+03] rad / s>,
+     'alfven_mode': <Quantity [7.48765909e-04, 2.13800404e+03] rad / s>,
+     'acoustic_mode': <Quantity [0.00043295, 0.07358991] rad / s>}
+    """
+
+    # validate argument ion
+    if not isinstance(ion, Particle):
+        try:
+            ion = Particle(ion)
+        except TypeError:
+            raise TypeError(
+                f"For argument 'ion' expected type {Particle} but got {type(ion)}."
+            )
+    if not (ion.is_ion or ion.is_category("element")):
+        raise ValueError("The particle passed for 'ion' must be an ion or element.")
+
+    # validate z_mean
+    if z_mean is None:
+        try:
+            z_mean = abs(ion.integer_charge)
```

Comment:
```suggestion
            z_mean = abs(ion.charge_number)
```

name `integer_charge` as dropped for `charge_number` in PR #1136